src/main/java/uk/gov/justice/hmpps/prison/api/resource/ImageResource.java

```diff
@@ -66,4 +72,19 @@
     public ImageDetail getImage(@PathVariable("imageId") @ApiParam(value = "The image id of offender", required = true) final Long imageId) {
         return imageService.findImageDetail(imageId);
     }
+
+    @ApiResponses({
+        @ApiResponse(code = 200, message = "OK", response = ImageDetail.class),
+        @ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse.class),
+        @ApiResponse(code = 404, message = "The offender number could not be found or has no bookings.", response = ErrorResponse.class),
+        @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse.class)})
+    @ApiOperation(value = "Upload a new image for an offender.", notes = "Image data is base64-encoded JPEG content. Requires ROLE_SYSTEM_USER.", nickname = "putImage", hidden = true)
+    @PutMapping("/offender/{offenderNo}")
+    public ImageDetail putImage(
+        @Pattern(regexp = "^[A-Z]\\d{4}[A-Z]{2}$", message = "Offender Number format incorrect") @PathVariable("offenderNo") @ApiParam(value = "The offender number relating to this image.", required = true) final String offenderNo,
+        @RequestParam(value = "fullSizeImage", defaultValue = "false") @ApiParam(value = "Set to true if this is a full size image", defaultValue = "false") final boolean fullSizeImage,
+        @RequestBody @ApiParam(value = "The base64-encoded string of the JPEG image data.", required = true) final String imageData
```

Comment:
could this be byte[] instead of String?

Reply:
It could, but is there much difference? Client is required to base64-encode the image and String might be a more universal type?